File: plugins/appvador/proj.ios/AdsAppvador.m
/****************************************************************************
 Copyright (c) 2013 cocos2d-x.org
 
 http://www.cocos2d-x.org
 
 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:
 
 The above copyright notice and this permission notice shall be included in
 all copies or substantial portions of the Software.
 
 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 ****************************************************************************/

#import "AdsAppvador.h"
#import "AdsWrapper.h"

#define OUTPUT_LOG(...)     if (self.debug) NSLog(__VA_ARGS__);

@implementation AdsAppvador
{
    AvAdView* avAdView;
}

@synthesize debug = __debug;

#pragma mark InterfaceAds impl

- (void) configDeveloperInfo: (NSMutableDictionary*) devInfo
{
    NSString* bannerAppId = (NSString*) [devInfo objectForKey:@"AppvadorBannerID"];
    UIViewController* rootViewController = [AdsWrapper getCurrentRootViewController];
    UIView* rootView = rootViewController.view;
    CGSize screenSize = rootView.frame.size;
    avAdView = [[AvAdView alloc] initWithFrame:CGRectMake(0, screenSize.height-kBannerHeight, kBannerWidth, kBannerHeight) applicationId:bannerAppId];
    avAdView.rootViewController = rootViewController;
    avAdView.delegate = self;
    [avAdView isTest:self.debug];
}

- (void) showAds: (NSMutableDictionary*) info position:(int) pos
{
    if (avAdView) {
        [[AdsWrapper getCurrentRootViewController].view addSubview:avAdView];
        [avAdView adStart];
    } else {
        [AdsWrapper onAdsResult:self withRet:kUnknownError withMsg:@"banner"];
    }
}

- (void) hideAds: (NSMutableDictionary*) info
{
    if ([avAdView superview]) {
        [avAdView removeFromSuperview];
    }
}

- (void) queryPoints
{
    OUTPUT_LOG(@"Appvador not support query points!");
}

- (void) spendPoints: (int) points
{
    OUTPUT_LOG(@"Appvador not support spend points!");
}

- (void) setDebugMode: (NSNumber*) isDebugMode
{
    self.debug = [isDebugMode boolValue];
}

- (NSString*) getSDKVersion
{
    return @"1.2.6";
}

- (NSString*) getPluginVersion
{
    return @"0.0.1";
}

//バナータップ時
- (void)avAdDidTap:(AvAdView*)avadview
{
    OUTPUT_LOG(@"AdsAppvador:avAdDidTap!");
}

//バナー広告ページが表示された時
- (void)avAdDidOpenFullMovieView:(AvAdView*)avadview
{
    OUTPUT_LOG(@"AdsAppvador:avAdDidOpenFullMovieView!");
    [AdsWrapper onAdsResult:self withRet:kAdsShown withMsg:@"banner"];
}

//バナー広告ページを閉じたとき
- (void)avAdDidCloseFullMovieView:(AvAdView*)avadview
{
    OUTPUT_LOG(@"AdsAppvador:avAdDidCloseFullMovieView!");
    [AdsWrapper onAdsResult:self withRet:kAdsDismissed withMsg:@"banner"];
}

//バナー広告読み込み完了時
- (void)avAdDidFinishedLoad:(AvAdView*)avadview
{
    OUTPUT_LOG(@"AdsAppvador:avAdDidFinishedLoad!");
}

//バナー広告読み込み失敗時
- (void)avAdDidFailToReceiveAd:(AvAdView*)avadview
{
    OUTPUT_LOG(@"AdsAppvador:avAdDidFailToReceiveAd!");
    //viewの削除
    if (avAdView) {
        [avAdView remove];
        avAdView = nil;
    }
    [AdsWrapper onAdsResult:self withRet:kUnknownError withMsg:@"banner"];
}

@end

File: plugins/appvador/proj.android/src/org/cocos2dx/plugin/AdsAppvador.java
/****************************************************************************
Copyright (c) 2012-2013 cocos2d-x.org

http://www.cocos2d-x.org

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in
all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
THE SOFTWARE.
****************************************************************************/
package org.cocos2dx.plugin;

import java.util.Hashtable;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.util.Log;
import android.util.DisplayMetrics;
import android.view.Gravity;
import android.view.View;
import android.view.ViewGroup;
import android.view.WindowManager;
import android.widget.LinearLayout;
import android.widget.RelativeLayout;

import com.appvador.ad.*;

public class AdsAppvador implements InterfaceAds {

	private static final String LOG_TAG = "AdsAppvador";
	private static Activity mContext = null;
	private static boolean bDebug = false;
    private static AdView _adView = null;
    private static String bannerAppId = "";
    private static RelativeLayout _adMain = null;
    private static AdsAppvador mAdapter = null;
    
	protected static void LogD(String msg) {
		if (bDebug) {
			Log.d(LOG_TAG, msg);
		}
	}

	public AdsAppvador(Context context) {
		mContext = (Activity) context;
        mAdapter = this;
	}

	@Override
	public void setDebugMode(boolean debug) {
		bDebug = debug;
	}

	@Override
	public void configDeveloperInfo(Hashtable<String, String> devInfo) {
        bannerAppId = devInfo.get("AppvadorBannerID");
	}
    
    public void removeAdView() {
        LogD("AppVador removeMainView!");
        
        _adView.stop();
        _adMain.removeView(_adView);
        _adView = null;
        
        Activity activity = mContext;
        View contentView = ((ViewGroup)activity.findViewById(android.R.id.content)).getChildAt(0);
        ((ViewGroup)contentView).removeView(_adMain);
        _adMain = null;
    }
    
    public void initAppvador() {
        LogD("AppVador hide!");
        
        Activity activity = mContext;
        _adView = new AdView(activity, bannerAppId, false, bDebug);
        _adView.setAdListener(new AppvadorAdsListener());
        
        _adMain = new RelativeLayout(activity);
        RelativeLayout.LayoutParams layoutParams = new RelativeLayout.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT);
        layoutParams.addRule(RelativeLayout.CENTER_HORIZONTAL);
        layoutParams.addRule(RelativeLayout.ALIGN_PARENT_BOTTOM);
        
        _adView.setVisibility(View.INVISIBLE);
        _adMain.addView(_adView, layoutParams);
        
        View contentView = ((ViewGroup)activity.findViewById(android.R.id.content)).getChildAt(0);
        ((ViewGroup)contentView).addView(_adMain);
    }

	@Override
	public void showAds(Hashtable<String, String> info, int pos) {
		PluginWrapper.runOnMainThread(new Runnable() {
			@Override
			public void run() {
                if (_adMain != null) {
                    removeAdView();
                }
                initAppvador();
                
                _adView.setVisibility(View.VISIBLE);
                _adView.adStart();
			}
		});
	}

	@Override
	public void hideAds(Hashtable<String, String> info) {
        PluginWrapper.runOnMainThread(new Runnable() {
			@Override
			public void run() {
                LogD("AppVador hideAds!");
                if (_adMain != null) {
                    LogD("AppVador hide!");
                    //削除
                    removeAdView();
                    //削除するとなぜか音が調整できなくなるので、initして非表示にしておく
                    initAppvador();
                }
			}
		});
	}

    @Override
	public void spendPoints(int points) {
		LogD("AppVador not support spend points!");
	}
    
    private class AppvadorAdsListener implements AdListener {
        
        @Override
        public void detachedFromWindow(AdResult args0) {
            LogD("AppVador detachedFromWindow!");
        }
        
        //広告の取得に成功
        @Override
        public void adLoadSucceeded(AdResult args0) {
            LogD("AppVador adLoadSucceeded!");
        }
        
        //広告の取得に失敗
        @Override
        public void failedToReceiveAd(AdResult args0) {
            LogD("AppVador failedToReceiveAd!");
            //削除
            _adView.stop();
            _adMain.removeView(_adView);
            _adView = null;
            
            Activity activity = mContext;
            View contentView = ((ViewGroup)activity.findViewById(android.R.id.content)).getChildAt(0);
            ((ViewGroup)contentView).removeView(_adMain);
            _adMain = null;
            
            AdsWrapper.onAdsResult(mAdapter, AdsWrapper.RESULT_CODE_UnknownError, "banner");
        }
        
        //広告がタップされた際に呼ばれます
        @Override
        public void adDidTap(AdResult args0) {
            LogD("AppVador adDidTap!");
        }
	}
    
    @Override
	public void queryPoints() {
		LogD("AppVador not support query points!");
	}
    
	@Override
	public String getSDKVersion() {
		return "1.1.1";
	}
    
	@Override
	public String getPluginVersion() {
		return "0.0.1";
	}
}
